client/src/main/java/org/apache/rocketmq/client/producer/RequestFutureHolder.java

```diff
@@ -74,14 +69,40 @@ public void scanExpiredRequest() {
         }
     }
 
-    private RequestFutureHolder() {
+    public synchronized void startScheduledTask() {
+        if (this.producerNum.incrementAndGet() == 1) {
+            this.getScheduledExecutorService().scheduleAtFixedRate(new Runnable() {
+                @Override
+                public void run() {
+                    try {
+                        RequestFutureHolder.getInstance().scanExpiredRequest();
+                    } catch (Throwable e) {
+                        log.error("scan RequestFutureTable exception", e);
+                    }
+                }
+            }, 1000 * 3, 1000, TimeUnit.MILLISECONDS);
+        }
+    }
+
+    public synchronized void shutdown() {
+        if (this.producerNum.decrementAndGet() == 0) {
+            this.scheduledExecutorService.shutdown();
```

Comment:
check null

Reply:
may thutdown() throws exception? 

the below code maybe better?
```
ScheduledExecutorService ses = this.scheduledExecutorService;
this.scheduledExecutorService = null;
ses.shutdown();
```

Reply:
yes，it would be better